File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
from math import gcd

class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1:
            return ""
        
        len1 = len(str1)
        len2 = len(str2)
        len_gcd = gcd(len1, len2)
        
        return str1[:len_gcd]

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.java
class Solution {
    public String gcdOfStrings(String str1, String str2) {
        if ((str1 + str2).equals(str2 + str1) == false) {
            return "";
        }
        
        int len1 = str1.length();
        int len2 = str2.length();
        int len_gcd = gcd(len1, len2);
        
        return str1.substring(0, len_gcd);
        
    }
    
    static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        
        return gcd(b, a % b);
    }
}